### PR TITLE
Fix cmtlog fork

### PR DESCRIPTION
**Pull request by kape1395**

### CODE REVIEW

File: packages/chain/chainMgr/chainMgr.go [hunk 1]
```diff
@@ -290,13 +291,15 @@ func (cmi *chainMgrImpl) handleInputChainTxPublishResult(input *inputChainTxPubl
 	delete(cmi.needPublishTX, input.txID)
 	if input.confirmed {
 		// >     If result.confirmed = false THEN ... ELSE
-		// >         NOP // AO has to be received as Confirmed AO.
-		return nil
+		// >         NOP // AO has to be received as Confirmed AO. // TODO: Not true, anymore.
```
Comment (jorgemmsilva) [inline packages/chain/chainMgr/chainMgr.go:294]: TODO still valid?

File: packages/chain/cmtLog/cmtLog.go [hunk 2]
```diff
@@ -244,95 +269,79 @@ func (cl *cmtLogImpl) Message(msg gpa.Message) gpa.OutMessages {
 }
 
 // > UPON AliasOutput (AO) {Confirmed | Rejected} by L1:
-// >     LocalView.Update(AO)
-// >     IF LocalView changed THEN
-// >         LogIndex.L1ReplacedBaseAliasOutput()
-// >         TryProposeConsensus()
+// >   ...
```
Comment (jorgemmsilva) [inline packages/chain/cmtLog/cmtLog.go:272]: many `// ...` comments

File: packages/chain/cmtLog/cmtLog_rapid_test.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+// 	sm.tc.RunAll()
+// }
+
+// func (sm *cmtLogTestRapidSM) ConsRejected(t *rapid.T) {
+// 	nodeID := sm.genNodeID.Draw(t, "node")
+// 	var ao *isc.AliasOutputWithID // TODO: Set it.
+// 	var li cmtLog.LogIndex        // TODO: Set it.
+// 	sm.tc.WithInput(nodeID, cmtLog.NewInputConsensusOutputRejected(ao, li))
+// 	sm.tc.RunAll()
+// }
```
Comment (jorgemmsilva) [inline packages/chain/cmtLog/cmtLog_rapid_test.go:132]: should these be removed?

File: packages/chain/cmtLog/cmtLog_rapid_test.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+// We want here to check liveness -- if the chain never stops. This test framework
+// don't allow to check temporal properties, only state predicates and invariants.
+// So we reformulate the property to the condition, that always, either TX is confirming
+// or consensus is running. Assuming fairness for both, it should imply liveness of
+// this algorithm.
+func (sm *cmtLogTestRapidSM) invHaveConsRunningOrTxConfirming(t *rapid.T) {
+	// TODO: >...
+}
```
Comment (jorgemmsilva) [inline packages/chain/cmtLog/cmtLog_rapid_test.go:172]: remove?

File: packages/chain/cmtLog/var_localView_rapid_test.go [hunk 5]
```diff
@@ -141,7 +141,7 @@ func (sm *varLocalViewSM) ConsensusOutput(t *rapid.T) {
 	prevAO := sm.lv.Value()
 	require.NotNil(t, prevAO)
 	newAO := sm.nextAO(prevAO)
-	tipAO, tipChanged := sm.lv.ConsensusOutputDone(prevAO.OutputID(), newAO)
+	tipAO, tipChanged := sm.lv.ConsensusOutputDone(cmtLog.NilLogIndex(), prevAO.OutputID(), newAO) // TODO: LogIndex.
```
Comment (jorgemmsilva) [inline packages/chain/cmtLog/var_localView_rapid_test.go:144]: TODO can be removed?

**Review by jorgemmsilva — APPROVED** [review]

File: packages/chain/chainMgr/chainMgr.go (hunk 1 above)
Comment (kape1395) [reply packages/chain/chainMgr/chainMgr.go:294]: Yes, still valid.

File: packages/chain/cmtLog/cmtLog.go (hunk 2 above)
Comment (kape1395) [reply packages/chain/cmtLog/cmtLog.go:272]: Incomplete test case. I will continue to work on it.

File: packages/chain/cmtLog/cmtLog_rapid_test.go (hunk 3 above)
Comment (kape1395) [reply packages/chain/cmtLog/cmtLog_rapid_test.go:132]: No, please keep them for now.

File: packages/chain/cmtLog/cmtLog_rapid_test.go (hunk 4 above)
Comment (kape1395) [reply packages/chain/cmtLog/cmtLog_rapid_test.go:172]: I will work on this test case further.

File: packages/chain/cmtLog/var_localView_rapid_test.go (hunk 5 above)
Comment (kape1395) [reply packages/chain/cmtLog/var_localView_rapid_test.go:144]: Not yet.

**Pull request merged**